py-polars/polars/internals/type_aliases.py

```diff
@@ -14,6 +16,11 @@
 else:
     from typing_extensions import TypeAlias
 
+from typing import Union
+
+# Types that qualify as expressions (eg: for use in 'select', 'with_columns'...)
+PolarsExprType: TypeAlias = "pli.Expr | pli.WhenThen | pli.WhenThenThen | pli.Series"
```

Comment:
`pli.Series` is technically not an expression, and gets passed to `lit` before usage. I would leave it out here for correctness, and specify it explicitly where appropriate.